controllers/spacebindingrequestmigration/spacebindingrequest_migration_controller.go

```diff
@@ -0,0 +1,185 @@
+	}
+
+	// skip workspace creator spacebinding
+	// the controller will convert only spacebindings created by system admins using the sandbox-cli.
+	// If the creator label on the space matches the usersignup reference on the MUR then this is the owner of the space
+	// and the spacebinding should not be migrated.
+	if mur.ObjectMeta.OwnerReferences == nil || len(mur.ObjectMeta.OwnerReferences) == 0 {
+		return ctrl.Result{}, errs.New("MasterUserRecord has no UserSignup owner reference")
+	}
+	// skip spaces with no creator label
+	if _, ok := space.Labels[toolchainv1alpha1.SpaceCreatorLabelKey]; !ok {
+		// let's log an error, since this should not happen in production
+		logger.Error(errs.New("space has no SpaceCreatorLabelKey set"), "the spacebindings for this space will not be migrated", "space name", space.Name)
+		return ctrl.Result{}, nil
+	}
+	usersignup := mur.ObjectMeta.OwnerReferences[0]
+	if space.Labels[toolchainv1alpha1.SpaceCreatorLabelKey] == usersignup.Name {
+		return reconcile.Result{}, nil
+	}
+
+	// get the spaceRole
+	spaceRole := spaceBinding.Spec.SpaceRole
+
+	// get member cluster name where the space was provisioned
+	targetCluster := space.Spec.TargetCluster
+	memberCluster, memberClusterFound := r.MemberClusters[targetCluster]
+	if !memberClusterFound {
+		return ctrl.Result{}, errs.New(fmt.Sprintf("unable to find member cluster: %s", targetCluster))
+	}
+
+	// get the home namespace from space
+	defaultNamespace := getDefaultNamespace(space.Status.ProvisionedNamespaces)
+
+	// construct a SpaceBindingRequest object
+	sbrName := mur.GetName() + "-" + spaceRole
+	sbr := &toolchainv1alpha1.SpaceBindingRequest{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      sbrName,
+			Namespace: defaultNamespace,
+			Labels: map[string]string{
+				toolchainv1alpha1.SpaceCreatorLabelKey:                 "spacebindingrequestmigration",
+				toolchainv1alpha1.SpaceBindingSpaceLabelKey:            space.GetName(),
+				toolchainv1alpha1.SpaceBindingMasterUserRecordLabelKey: mur.GetName(),
+			},
+		},
+		Spec: toolchainv1alpha1.SpaceBindingRequestSpec{
+			MasterUserRecord: mur.GetName(),
+			SpaceRole:        spaceRole,
+		},
+	}
+
+	// check if SpaceBindingRequest doesn't exist already
+	err = memberCluster.Client.Get(context.TODO(), types.NamespacedName{
+		Namespace: defaultNamespace,
+		Name:      sbrName,
+	}, &toolchainv1alpha1.SpaceBindingRequest{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// create the sbr resource
+			if err := memberCluster.Client.Create(ctx, sbr); err != nil {
+				return ctrl.Result{}, errs.Wrapf(err, "unable to create SpaceBindingRequest")
+			}
+
+			// delete the SpaceBinding. The SpaceBindingRequest controller will create a new SpaceBinding for the SBR we created above.
+			if err := r.Client.Delete(ctx, spaceBinding); err != nil {
+				return ctrl.Result{}, errs.Wrapf(err, "unable to delete the SpaceBinding")
+			}
+			return ctrl.Result{}, nil
```

Comment:
this should be executed when the SBR is already present. The controller watches both SBs and SBRs, so you can create the SBR and then just return and wait for the next reconcile triggered by the creation. 

This will also cover the situation when the deletion call fails for any reason - I'm talking abous a situation:
1. SpaceBinding is not migrated
2. the controller creates SBR CR
3. the `Delete` call fails for any reason
4. the controller returns and error
5. in the next reconcile, the controller doesn't get to this code because the SBR is already present.

Reply:
I'm sorry, but If I'm not wrong, this way I'll be deleting the SpaceBinding owned by the SBR and not the one that was migrated right? 

I mean , if I defer the Delection of the SpaceBinding only when there's a SpaceBindingRequest associated to it, I'll not be deleting the intended objects , but I'm sure I'm missing something here 🤔

Reply:
If the SpaceBinding is already managed by SBR, then you shouldn't get to this point - you return from the reconcile much earlier: 
https://github.com/codeready-toolchain/host-operator/blob/f5ff6c7540fb694fd43b2403da30b589cac81595/controllers/spacebindingrequestmigration/spacebindingrequest_migration_controller.go#L71-L75

Reply:
As discussed, I have:
- forced a requeue after the creation of the SBR 
- moved the deletion of the migrated SpaceBinding object at the end, only after we checked that the SBR was already created

please check https://github.com/codeready-toolchain/host-operator/pull/922/commits/bf6f19d1839abd949dfda81ec4cf54c0f792227f

Thanks for your suggestion!